pkg/controller/traceflow/controller_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceflow
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+
+	ops "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	fakeversioned "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned/fake"
+	crdinformers "github.com/vmware-tanzu/antrea/pkg/client/informers/externalversions"
+)
+
+var alwaysReady = func() bool { return true }
+
+const informerDefaultResync time.Duration = 30 * time.Second
+
+type traceflowController struct {
+	*Controller
+	client             versioned.Interface
+	informerFactory    informers.SharedInformerFactory
+	crdInformerFactory crdinformers.SharedInformerFactory
+}
+
+func newController() *traceflowController {
+	client := fake.NewSimpleClientset()
+	crdClient := fakeversioned.NewSimpleClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, informerDefaultResync)
+	crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClient, informerDefaultResync)
+	controller := NewTraceflowController(crdClient,
+		informerFactory.Core().V1().Pods(),
+		crdInformerFactory.Ops().V1alpha1().Traceflows())
+	controller.traceflowListerSynced = alwaysReady
+	return &traceflowController{
+		controller,
+		crdClient,
+		informerFactory,
+		crdInformerFactory,
+	}
+}
+
+func TestTraceflow(t *testing.T) {
+	tfc := newController()
+	assert.NotNil(t, tfc)
+
+	stopCh := make(chan struct{})
+	go tfc.Run(stopCh)
+	time.Sleep(2 * time.Second)
+
+	tf1 := ops.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{Name: "tf1", UID: "uid1"},
+		Spec:       ops.TraceflowSpec{},
+	}
+	tfc.client.OpsV1alpha1().Traceflows().Create(context.TODO(), &tf1, metav1.CreateOptions{})
+	tfc.addTraceflow(&tf1)
```

Comment:
This will trigger Traceflow controller to process the change, but Lister cannot find "tf1" in syncTraceflow().

@tnqn: Anything wrong?

Reply:
Perhaps it's because informerFactory and crdInformerFactory is not started, try
```
tfc.informerFactory.Start(stopCh)
tfc.crdInformerFactory.Start(stopCh)
```

Reply:
Ok. It works!